resources/fetchers/ecr_fetcher.go

```diff
@@ -114,12 +117,11 @@ func (res ECRResource) GetData() interface{} {
 }
 
 func (res ECRResource) GetMetadata() fetching.ResourceMetadata {
-	uid, _ := uuid.NewV4()
 	return fetching.ResourceMetadata{
-		ID:      uid.String(),
-		Type:    fetching.ECRType,
-		SubType: fetching.ECRType,
-		Name:    "AWS repositories",
+		ID:      *res.RepositoryArn,
+		Type:    ecrResourceType,
+		SubType: ecrResourceSubType,
+		Name:    *res.RepositoryName,
```

Comment:
is there a risk of null dereference here?

Reply:
Most likely not because it means that no error was received while fetching the AWS resource and the response is missing/ partly missing - but better be safe than sorry.